.pre-commit-config.yaml

```diff
@@ -0,0 +1,18 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: check-docstring-first
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+  - repo: https://github.com/myint/autoflake
+    rev: v1.4
+    hooks:
+      - id: autoflake
+        args: ["--in-place", "--remove-all-unused-imports", "--ignore-init-module-imports", "--remove-unused-variables"]
```

Comment:
I don't like auto-corrections that go beyond basic stuff like end-fo-file-fixer or trailing-whitespace. Anyway, one cannot apply the unused import detection as is to the current codebase because we have cases like this:

https://github.com/matplotlib/matplotlib/blob/c12791c29c4785dbead2bf16ede15696d27a7f69/lib/matplotlib/text.py#L16

So let's leave this out.

Reply:
Sure thing.

Although note that I think this basically pushes you to to use `__all__` or `# noqa` both of which would prevent that from being removed.

Reply:
We currently do that via F401 excludes in `.flake8`.

The `# noqa` comment is quite non-instructive. Technically, something like would work for flake8 but has the disadvantage that lines become too long very quickly and you get line-length-complaints. Also, last time I checked, not all tools can work with this expanded syntax.
```
from .textpath import TextPath  # Unused, but imported by others. # noqa: F401
```

Reply:
I don't think autocorrections are a good idea either.  That seems likely to sow more confusion than it helps.  I don't think new contributors find flake8 intimidating, and many IDEs will show you where the flake8 errors are already, so I don't think its too cumbersome to ask users to edit by hand.  If this were togglable, maybe that would be OK, but default off?

Reply:
> The # noqa comment is quite non-instructive. 

Agreed, my preference would still be to use `__all__ = ["TextPath", ...]` and re-enable F401. This feels like the correct way to specify the "exports" of a file while also allowing checks for actually unused imports.